code/lib/builder-vite/src/plugins/csf-plugin.ts

```diff
@@ -0,0 +1,15 @@
+import type { Plugin } from 'vite';
+import { vite } from '@storybook/csf-plugin';
+import type { StorybookConfig } from '@storybook/types';
+import type { ExtendedOptions } from '../types';
+
+export async function csfPlugin(config: ExtendedOptions): Promise<Plugin> {
+  const { presets } = config;
+
+  const addons = await presets.apply<StorybookConfig['addons']>('addons', []);
+  const docsOptions =
+    // @ts-expect-error - not sure what type to use here
+    addons.find((a) => [a, a.name].includes('@storybook/addon-docs'))?.options ?? {};
+
+  return vite(docsOptions?.csfPluginOptions);
```

Comment:
It feels to me like `@storybook/csf-plugin` itself should take care of this...

Reply:
I don't think csf-plugin should know about the user's main.js and it makes sense for addon-docs to glue things together like this. Hopefully this all goes away in 7.x with the annotations server & simplifies things even further.